autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -0,0 +1,351 @@
+import logging
+import time
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+
+import torch
+from torch.autograd import Variable
+from torch.utils.tensorboard.writer import SummaryWriter
+
+from autoPyTorch.pipeline.components.training.base_training import autoPyTorchTrainingComponent
+
+
+class BudgetTracker(object):
+    def __init__(self,
+                 budget_type: str,
+                 max_value: int,
+                 ):
+        """
+        An object for tracking when to stop the network training.
+        It handles epoch based criteria as well as training based criteria.
+
+
+        # Maybe it is better to use pynisher? Remember the whole everything gets
```

Comment:
I would prefer to use pynisher, Matthias and I have ownership of that now

Reply:
Understood. I have added pynisher support both for memory and for runtime